server/server/controllers/customer.js

```diff
@@ -186,12 +186,52 @@ async function deleteCart(req, res) {
   }
 }
 
+async function checkout(req, res) {
+  const customerId = req.user.userId;
+  try {
+    //get the customer and fill the basket with the items data
+    const customer = await customerModel
+      .findById(customerId)
+      .populate("basket.items.itemId");
+    const basketItems = customer.basket.items;
+    if (basketItems.length < 1) {
+      return res.status(301).json({ message: "basket is empty!" });
```

Comment:
This should be handled in frontend phase where we redirect the user to homepage instead